src/isomorphic/classic/element/ReactElement.js

```diff
@@ -127,6 +127,10 @@ ReactElement.createElement = function(type, config, children) {
 
   if (config != null) {
     if (__DEV__) {
+      warning(
+        !(config.__proto__ != null && config.__proto__ !== Object.prototype),
```

Comment:
Let’s avoid double negations. You want to specify the “normal” condition in the `warning` call. The normal condition is when the `__proto__` ain’t there or it is `Object.prototype`. So let’s write it like

```js
 warning(
  config.__proto__ == null || config.__proto__ === Object.prototype,
```

instead.